velox/expression/Expr.cpp

```diff
@@ -1268,7 +1268,11 @@ void Expr::evalAll(
   // Write non-selected rows in remainingRows as nulls in the result if some
   // rows have been skipped.
   if (mutableRemainingRows != nullptr) {
-    addNulls(rows, mutableRemainingRows->asRange().bits(), context, result);
+    bool KnownToBeAllSelected = mutableRemainingRows->isAllSelectedSet() &&
```

Comment:
The new SelectivityVector::isAllSelectedSet method seems to exposed to much of implementation details of the SelectivityVector. I wonder if there is a cleaner way to implement the desired optimization. How do we end up with non-null mutableRemainingRows that have no nulls? Is this because decoded->nulls() above returns non-null even thought there are no null? Can we fix that instead?

Reply:
in the case i was looking at, the input is all not null with the null buffer allocated .
but in general this can happen whenever one of the inputs ended up a result with all not nulls 
but null buffer allocated. 

I wonder if we shall call allSelected() always instead. i am not sure much overhead that would add, but I do not think its high because we already traverse the nulls multiple times for every input anyway. in deselectNulls.